### ISSUE TITLE
GitHub Actions error collection script only reads latest attempt

### ISSUE THREAD

**Issue by timmc-edx** [edx/edx-arch-experiments#437]
A/C:

* The script iterates over all attempts rather than all jobs (we'll have to figure out how to do this)
* Run the script and verify that it reveals more connection lost errors than it did previously
* Write up brief findings on how often lost connection errors have been happening as a comment on https://github.com/openedx/edx-platform/issues/32671

The [Actions error collection script](https://github.com/edx/edx-arch-experiments/blob/main/edx_arch_experiments/scripts/get_action_errors.py) only collects the status of the most recent attempt on each job. Since we re-run most of our failed jobs, this script can't see most of the error information we're interested in.

See https://github.com/openedx/edx-platform/issues/32671 for an example of where this information would have been useful.

While we're in there, it might also be useful to turn this into a multi-stage script with caching. Currently there's a risk of getting rate-limited partway through a run, at which point all of the in-memory collected information is lost. It might be better to split the script so that it first gets all of the commits in the desired time range, writes that to file, and then gets job and attempt information -- but only for jobs that it hasn't already cached on disk. This would speed up future runs.

**Comment by RafayGhafoor** [comment]
@robrap @timmc-edx, I would like to work on this task and I am thinking of using [pyGithub](https://pygithub.readthedocs.io) library for the integration. Please let me know if I can work on this task.

**Comment by robrap** [comment]
@RafayGhafoor: That sounds good. Good luck.

**Comment by rgraber** [comment]
@RafayGhafoor are you still working on this?

**Comment by robrap** [comment]
@feanil: This might be a useful ticket for the Maintenance WG as well, because it would give you a view of issues across PRs to help with prioritization of tickets like https://github.com/edx/edx-arch-experiments/issues/528.

**Comment by timmc-edx** [comment]
I made an attempt at fixing this in <https://github.com/edx/edx-arch-experiments/pull/544>, which also includes some other improvements. But... it turns out all of the attempt objects for a workflow run are pointing to the same check suite! (The most recent one, naturally -- which means we lose any errors that provoke someone to re-run their tests.) This is blocked unless we can find a solution. I've posted about the issue at <https://github.com/orgs/community/discussions/103026>.
